### PR TITLE
[BACK-16] Manual addition of repositories to truffle

### PR DESCRIPTION
**Pull request by thomas-woehrle**
### Description of Issue
FR8 says:
_I can manually add entities, so Truffle AI can help me track them along with the most relevant ones
(e.g. when Truffle AI does not spot it automatically)._
### How I implemented
- adapted the schema
- added resolvers
- let chatGPT create a function for transforming the url into name and owner and fine-tuned that function
- tested
### Other
- in the resolvers i used unknown type and skipped the first argument passed to the resolver function
- Can be accessed via: 
`mutation {
   addProjectByName(name: "name", owner: "owner")
}`
- during implementation small issues with the current version of dbUpdater got apparent. While fix those in the future

### CODE REVIEW

**Review by pczern — CHANGES_REQUESTED** [review]

File: packages/graphql_server/src/graphql/resolvers.ts
[the remark this answers is not in the document]
Comment (cnk1dev) [reply packages/graphql_server/src/graphql/resolvers.ts:1]: @pczern idk is that usually done?

**Review by cnk1dev — APPROVED** [review]
Should be good to go after addressing @pczern's comment

File: packages/graphql_server/src/graphql/resolvers.ts [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
     helloWorld: () => 'Hello world!'
+  },
+  Mutation: {
+    // takes in variables. Parent object _ is never used
+    addProjectByName: async (_: unknown, { name, owner }: { name: string; owner: string }) => {
+      return await insertProject(name, owner, null)
+    },
+    // takes in variables. Parent object _ is never used
+    addProjectByUrl: async (_: unknown, { url }: { url: string }) => {
+      const urlParts = parseGitHubUrl(url)
```
[the remark this answers is not in the document]
Comment (thomas-woehrle) [reply packages/graphql_server/src/graphql/resolvers.ts:14]: @pczern Let's keep it simple for now -> true if inserting works, false if it doesn't

[the remark this answers is not in the document]
Comment (thomas-woehrle) [reply packages/graphql_server/src/graphql/resolvers.ts:14]: @pczern what do you mean with "check for all parts" ?

**Comment by thomas-woehrle** [comment]
@pczern anything new regarding this PR ? I responded to your comment

File: packages/graphql_server/src/graphql/resolvers.ts
[the remark this answers is not in the document]
Comment (pczern) [reply packages/graphql_server/src/graphql/resolvers.ts:1]: Not sure about that. But it makes sense to have a central place to decide which error messages are getting exposed, the resolvers here are a way to do this. If we didn't catch an error earlier, we might be exposing data we don't want to expose. More if we catched an error earlier, we might not have propagated the underlying error message; but a more generic message that makes debugging hard.

**Review by pczern — APPROVED** [review]

**Pull request merged**